ykcompile/src/lib.rs

```diff
@@ -1466,7 +1498,16 @@ impl<TT> TraceCompiler<TT> {
                         ; jne ->guardfail
                     );
                 }
-                Location::Mem(..) => todo!(),
+                Location::Mem(ro) => {
+                    debug_assert!(SIR.ty(&val.ty()).size() <= 8);
```

Comment:
That `8` is a bit of a magic number. Shall we make a const for it?

Reply:
Oh, that's left over from our debugging sessions. I don't think we need this here.